dartagnan/src/test/java/com/dat3m/dartagnan/spirv/SpirvHeaderTest.java

```diff
@@ -0,0 +1,520 @@
+package com.dat3m.dartagnan.spirv;
+
+import com.dat3m.dartagnan.exception.ParsingException;
+import com.dat3m.dartagnan.expression.IValue;
+import com.dat3m.dartagnan.parsers.program.ParserSpirv;
+import com.dat3m.dartagnan.program.Program;
+import com.dat3m.dartagnan.program.memory.Location;
+import com.dat3m.dartagnan.program.memory.MemoryObject;
+import com.dat3m.dartagnan.program.specification.AssertBasic;
+import com.dat3m.dartagnan.program.specification.AssertCompositeAnd;
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.CharStreams;
+import org.junit.Test;
+
+import java.util.Optional;
+
+import static org.junit.Assert.fail;
+
+public class SpirvHeaderTest {
```

Comment:
A lot of the tests in this class seem repeated with small modifications. Is there a way to factor the repeated parts away?

Reply:
Rewrote, is it better now?